pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/channel/MailboxContentStreamObserver.java

```diff
@@ -148,6 +148,7 @@ public void onError(Throwable e) {
   @Override
   public void onCompleted() {
     _isCompleted.set(true);
+    _gotMailCallback.accept(_mailboxId);
```

Comment:
shouldn't we also wake up onError as well. we need also to populate errors upstream?

Reply:
We are doing that already right? (L145 above)

Reply:
hmm. interesting i see some errors not populating back correctly issue. maybe it is a different problem (remove error vs. local error?)

Reply:
One open issue right now is that in case of an error we can potentially return "CANCELLED" error instead of the actual error block due to the race condition we talked about in the past. Are you seeing something else?

Reply:
Nope for #10555 the issue was with the sorting logic in the mailbox receive operator.